source/runtime.cpp

```diff
@@ -167,7 +174,12 @@ void reshade::runtime::on_present()
 			_effects_enabled = !_effects_enabled;
 
 		if (_input->is_key_pressed(_screenshot_key_data))
-			save_screenshot();
+		{
+			if (_screenshot_save_before)
+				_should_save_screenshot = true;
```

Comment:
This will cause the screenshots to mismatch because they are created one frame apart. Better to move the whole logic into `update_and_render_effects` and avoid having a member field altogether.
```c++
void reshade::runtime::update_and_render_effects()
{
    ...
    const auto input_lock = _input->lock();

    const auto do_screenshot = _input->is_key_pressed(_screenshot_key_data));
    if (do_screenshot && (_screenshot_save_before || !_effects_enabled))
        save_screenshot(_effects_enabled ? "before" : std::string());

    if (!_effects_enabled)
        return;
    ...
    if (do_screenshot)
        save_screenshot(_screenshot_save_before ? "after" : std::string());
}
```